### PR TITLE
Add risc-v support

### PR DESCRIPTION
**Pull request by diondokter**
I think this is all that's needed to get riscv support in this project.
Not sure this is the best panic instruction though...

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -12,11 +12,20 @@
...
+    not(any(feature = "cortex-m", feature = "risc-v")),
+    all(not(test), feature = "panic-handler")
+))]
+compile_error!("Enable either the cortex-m or risc-v feature");
+
 #[cfg(all(not(test), feature = "panic-handler"))]
 #[panic_handler]
 fn panic(_info: &core::panic::PanicInfo) -> ! {
     unsafe {
+        #[cfg(feature = "cortex-m")]
```
Comment (Yatekii) [inline src/lib.rs]: Why not use `target_arch `? :)

**Comment by Yatekii** [comment]
@diondokter should we try and move this forward? :)

**Comment by diondokter** [comment]
Ha well, I never got it to work and I kinda gave up on the chip. Was a small hobby project...
I'll use your suggestion, but I don't know if it actually works...

**Comment by diondokter** [comment]
@Yatekii Done, this is indeed better. Or should it do rv64 too?

File: Cargo.toml [hunk 2]
```diff
@@ -15,3 +15,5 @@ default = ["erase-chip", "panic-handler"]
 erase-chip = []
 panic-handler = []
 verify = []
+cortex-m = []
+risc-v = []
```
Comment (Yatekii) [inline Cargo.toml]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -27,8 +27,8 @@ jobs:
         cargo install cargo-binutils
         rustup component add llvm-tools-preview
     - name: Check
-      run: cargo check --target thumbv7em-none-eabi
+      run: cargo check --features cortex-m --target thumbv7em-none-eabi
     - name: Clippy
-      run: cargo clippy --target thumbv7em-none-eabi
+      run: cargo clippy --features cortex-m --target thumbv7em-none-eabi
```
Comment (Yatekii) [inline .github/workflows/ci.yml]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -27,8 +27,8 @@ jobs:
         cargo install cargo-binutils
         rustup component add llvm-tools-preview
     - name: Check
-      run: cargo check --target thumbv7em-none-eabi
+      run: cargo check --features cortex-m --target thumbv7em-none-eabi
```
Comment (Yatekii) [inline .github/workflows/ci.yml]:
```suggestion
```

**Comment by Yatekii** [comment]
Nice, thanks a lot! Let's add RV64 as well :)

**Comment by diondokter** [comment]
Ok, should be ready now!

**Review by Yatekii — APPROVED** [review]

**Pull request merged**